api/types/image/opts.go

```diff
@@ -27,7 +31,7 @@ type PullOptions struct {
 	// privilege request fails.
 	//
 	// Also see [github.com/docker/docker/api/types.RequestPrivilegeFunc].
-	PrivilegeFunc func() (string, error)
+	PrivilegeFunc func(context.Context) (string, error)
```

Comment:
Since it's a breaking change, should we first deprecate the old one, at least for one API version?

```suggestion
    // Deprecated: use [PrivilegeCtxFunc] instead
	PrivilegeFunc func() (string, error)
	
	PrivilegeCtxFunc func(context.Context) (string, error)
```

Then on the usage side, if `PrivilegeCtxFunc` is nil, call `PrivilegeFunc`.

Reply:
Was looking as well at this; I know in most cases we go to the deprecate (and alias) -> remove cycle. Not sure if we always were able to do this though, and it's possible we have changed signatures for other cases (thinking of context passing etc); at least the change wouldn't go "silent".

Tricky one

Reply:
Yeah, that makes sense; we're probably too far into RCs for this change.